### PR TITLE
modified set path operation to preserve query string

### PR DESCRIPTION
**Pull request by tnawathe21**
This PR modifies the set path operation to preserve the query string and rewrite only the main path. The code was verified with the test plan from previous PRs (see #9).

### CODE REVIEW

**Comment by tnawathe21** [comment]
Current dependencies on/for this PR:
* main
  * **PR #18** <a href="https://app.graphite.dev/github/pr/DataDog/envoy-filter-example/18" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/DataDog/envoy-filter-example/18?utm_source=stack-comment).

**Review by wdauchy — APPROVED** [review]

File: header-rewrite-filter/header_processor.cc [hunk 1]
```diff
@@ -149,8 +149,20 @@ namespace HeaderRewriteFilter {
         // cast to RequestHeaderMap because setPath is only on request side
         Http::RequestHeaderMap* request_headers = static_cast<Http::RequestHeaderMap*>(&headers);
         
-        // set path (path being set here includes the query string)
-        request_headers->setPath(request_path); // should never return an error
+        // get path (includes query string)
+        absl::string_view path = request_headers->path();
+
+        const size_t offset = path.find_first_of("?"); // TODO: envoy implements this as ?#
```
Comment (henrymwang) [inline header-rewrite-filter/header_processor.cc]: Could you use https://github.com/envoyproxy/envoy/blob/main/source/extensions/http/header_validators/envoy_default/path_normalizer.cc#L427 instead?

Comment (tnawathe21) [reply header-rewrite-filter/header_processor.cc]: It looks like the [visibility for this function](https://github.com/envoyproxy/envoy/blob/main/source/extensions/http/header_validators/envoy_default/BUILD#L48-L57) is not public

Comment (henrymwang) [reply header-rewrite-filter/header_processor.cc]: ack

**Review by henrymwang — APPROVED** [review]

**Pull request merged**